libraries/AP_Mount/AP_Mount_CADDX.cpp

```diff
@@ -137,8 +137,19 @@ void AP_Mount_CADDX::send_target_angles(const MountTarget& angle_target_rad)
 
     // byte 2's lower 3 bits are mode
     // lower 5 bits are sensitivity but always left as zero
-    uint8_t mode = (uint8_t)LockMode::TILT_LOCK | (uint8_t)LockMode::ROLL_LOCK;
-    if (angle_target_rad.yaw_is_ef) {
+    uint8_t mode = 0;
+    if (!option_set(Options::FPV_LOCK_DEFAULT)) {
+        mode = (uint8_t)LockMode::TILT_LOCK | (uint8_t)LockMode::ROLL_LOCK; // set rp ef defaults to RP ef, if option is not set
+    }
+    if (rc().find_channel_for_option(RC_Channel::AUX_FUNC::MOUNT_RP_LOCK)) { // if the rp lock aux sw is present use the state of roll and pitch lock variables set by it, or anyone else
```

Comment:
This should be done higher up in the AP_Mount_CADDX::update() method and only when the mode is, "MAV_MOUNT_MODE_RC_TARGETING".  If we do it down here then it will mean the gimbal will point in strange directions if, for example, the user selects, "Point Camera Here" from the MP Map